mediapipe/graphs/instantmotiontracking/calculators/sticker_manager_calculator.cc

```diff
@@ -0,0 +1,184 @@
+#include "mediapipe/framework/port/ret_check.h"
+#include "mediapipe/graphs/instantmotiontracking/calculators/transformations.h"
+
+namespace mediapipe {
+
+constexpr char kStringTag[] = "STRING";
+constexpr char kAnchorsTag[] = "ANCHORS";
+constexpr char kUserRotationsTag[] = "USER_ROTATIONS";
+constexpr char kUserScalingsTag[] = "USER_SCALINGS";
+constexpr char kRenderDescriptorsTag[] = "RENDER_DATA";
+
+// This calculator takes in the sticker data from the application as a string
+// in a sequence of the following format:
+// "(sticker_id:1,sticker_anchor_x:1.44,sticker_anchor_y:0.0,
+// sticker_rotation:0.0,sticker_scaling:0.0,sticker_render_id:0,
+// should_reset_anchor:true)(sticker_id:2.............."
+//
+// Input:
+//  STRING - String of sticker data in appropriate parse format
+// Output:
+//  ANCHORS - Anchors with initial normalized X,Y coordinates
+//  USER_ROTATIONS - UserRotations with radians of rotation from user
+//  USER_SCALINGS - UserScalings with increment of scaling from user
+//  RENDER_DATA - Descriptors of which objects/animations to render for stickers
+//
+// Example config:
+// node {
+//   calculator: "StickerManagerCalculator"
+//   input_stream: "STRING:sticker_data_string"
+//   output_stream: "ANCHORS:initial_anchor_data"
+//   output_stream: "USER_ROTATIONS:user_rotation_data"
+//   output_stream: "USER_SCALINGS:user_scaling_data"
+//   output_stream: "RENDER_DATA:sticker_render_data"
+// }
+
+class StickerManagerCalculator : public CalculatorBase {
+ public:
+  static ::mediapipe::Status GetContract(CalculatorContract* cc) {
+    RET_CHECK(!cc->Inputs().GetTags().empty());
+    RET_CHECK(!cc->Outputs().GetTags().empty());
+
+    if (cc->Inputs().HasTag(kStringTag)) {
+      cc->Inputs().Tag(kStringTag).Set<std::string>();
+    }
+    if (cc->Outputs().HasTag(kAnchorsTag)) {
+      cc->Outputs().Tag(kAnchorsTag).Set<std::vector<Anchor>>();
+    }
+    if (cc->Outputs().HasTag(kUserRotationsTag)) {
+      cc->Outputs().Tag(kUserRotationsTag).Set<std::vector<UserRotation>>();
+    }
+    if (cc->Outputs().HasTag(kUserScalingsTag)) {
+      cc->Outputs().Tag(kUserScalingsTag).Set<std::vector<UserScaling>>();
+    }
+    if (cc->Outputs().HasTag(kRenderDescriptorsTag)) {
+      cc->Outputs()
+          .Tag(kRenderDescriptorsTag)
+          .Set<std::vector<RenderDescriptor>>();
+    }
+
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Open(CalculatorContext* cc) final {
+    cc->SetOffset(TimestampDiff(0));
+    return ::mediapipe::OkStatus();
+  }
+
+  ::mediapipe::Status Process(CalculatorContext* cc) final {
+    std::string sticker_data_string =
+        cc->Inputs().Tag(kStringTag).Get<std::string>();
+
+    std::vector<Anchor> initial_anchor_data;
+    std::vector<UserRotation> user_rotation_data;
+    std::vector<UserScaling> user_scaling_data;
+    std::vector<RenderDescriptor> render_descriptor_data;
+
+    while (sticker_data_string.find(")") != -1) {
```

Comment:
After seeing this, I'd highly recommend (for the following pull request) that we convert this over to protobuffers-- the string-based approach will just get tougher and tougher to maintain.